### ISSUE TITLE
Custom editor icons should change color with light themes

### ISSUE THREAD

**Issue by aaronfranke** [godotengine/godot-proposals#572]
**Describe the project you are working on:** [2.5D demo project](https://github.com/godotengine/godot-demo-projects/tree/master/misc/2.5d), and anything else with editor icons.

**Describe the problem or limitation you are having in your project:**

Right now the main screen plugin icon looks fine with the editor's default theme:

![Screenshot from 2020-03-10 09-12-18](https://user-images.githubusercontent.com/1646875/76315767-bdb86a80-62af-11ea-81e1-1dcdfdd4b5c7.png)

But it doesn't look right with the "Light" theme, the other icons are dark but 2.5D remains light and therefore becomes invisible against the light background:

![Screenshot from 2020-03-10 09-12-09](https://user-images.githubusercontent.com/1646875/76315773-bf822e00-62af-11ea-84cb-51265cca776f.png)

This problem also exists to a lesser extent in the hierarchy, where the purple Node25D icon really should be a darker shade similarly to how the Node2D/etc icons become a darker shade:

![3](https://user-images.githubusercontent.com/1646875/76315840-df195680-62af-11ea-8f28-262c191012c3.png)

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

The proposed enhancement is that, whatever logic is applied to change the color of editor icons when a light theme is activated, the same logic should also be applied to user-created custom icons when they're displayed in the editor. See my comment [here](https://github.com/godotengine/godot/pull/14348#issuecomment-593829129).

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

I'm not exactly sure how to current logic works. Does it just change the brightness? The code would be the same as how the current logic works.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

There doesn't seem to be a way to get the calculated value of whether the icon and font color is light or dark. If you run the following code in an EditorPlugin script, it prints `0` (automatic):

    print(get_editor_interface().get_editor_settings().get_setting("interface/theme/icon_and_font_color"))

The best solution for now is to have the above code (in case users wish to specifically enforce light/dark icon/font color) combined with code like this:

    print(get_editor_interface().get_editor_settings().get_setting("interface/theme/base_color").v > 0.5)

However, this doesn't update the plugin unless the plugin is restarted, and it also may not be consistent with the code that the editor uses to determine whether to use light or dark icons and font color. Therefore, this can't properly be worked around.

**Is there a reason why this should be core and not an add-on in the asset library?:**

This can't be worked around properly in an add-on, and it is about making add-ons better.

**Comment by Jummit** [comment]
I don't think it just changes the colors, since many icons don't work in bright mode. See the 2.5D icon in the image in OP.

**Comment by Calinou** [comment]
I see two ways to do this:

1. Always darken custom node icons when using a light theme by a preset factor (like 20-30%). This is easy to do, but it won't look identical to the default icons if you use the default colors.
2. Expose the [icon color conversion table](https://github.com/godotengine/godot/blob/3c376a898ae6f4b6332d732d29c44d090978b887/editor/editor_themes.cpp#L108-L256) somehow and use it to replace colors in SVG icons before rendering them. This is more complex and can't be done reliably for PNG icons (due to antialiasing being baked into the icons).

**Comment by Jummit** [comment]
For svg icons I would definitely love number two.

**Comment by groud** [comment]
>  Unfortunately, this can't be done reliably for PNG icons since antialiasing is often baked into the icons. This is also more complex compared to the first solution I mentioned.

Think it's probably possible to do something like that with something similar to a LUT. 
I believe it could be possible to compute the a new color for all colors using a set of provided mapping, and interpolation between those mapping.

**Comment by Calinou** [comment]
@groud If we're going for a complex solution, I'd rather just tell all plugin authors to switch to SVG icons :slightly_smiling_face:

**Comment by pycbouh** [comment]
Due to complexity of this problem, I think this should be on plugin developers to handle dark theme, but the engine should provide tools to make their life easier.

1. The color conversion methods from `ImageLoaderSVG` must be exposed in some form to allows similar mechanics as the editor uses. That means we need a method to add to the color mapping list and the method for conversion itself. Or, because nobody is probably loading `.svg` files by hand this should be an option that can be setup before loading the file.
2. For rasterized images we can only provide recommendations. While there can be a calculated value exposed in editor theme for that purpose, I'm not sure if it's going to be reliable enough.
3. Overall, there needs to be an article in documentation explaining the best practices on how to make plugins integrated into the editor in the best possible way. That includes this problem with icons, usage of editor styles (colors, fonts, etc), and recommendations about what tools are better placed in what containers.

**Comment by adamscott** [comment]
I suggest a simpler path for developers. [The technique used by VS Code extensions is simple enough:](https://code.visualstudio.com/api/references/contribution-points#contributes.commands)

```json
{
  "contributes": {
    "commands": [
      {
        "command": "extension.sayHello",
        "title": "Hello World",
        "category": "Hello",
        "icon": {
          "light": "path/to/light/icon.svg",
          "dark": "path/to/dark/icon.svg"
        }
      }
    ]
  }
}
```

For Godot, if only one icon is supplied, the current behavior does not change. But if two icons are supplied, then the first one is for light themes, and the second one is for dark themes. (or vice-versa)

```gdscript
# Default behavior
class_name Anchors, "res://icon-light.svg", "res://icon-dark.svg"

# Single icon should continue to work
class_name Anchors, "res://icon.svg"
```

**Comment by aaronfranke** [comment]
> But if two icons are supplied, then the first one is for light themes, and the second one is for dark themes. (or vice-versa)

Considering the default theme is a dark theme, vice versa would make sense.
